packages/xo-server/src/xo-mixins/proxies.js

```diff
@@ -163,6 +163,13 @@ export default class Proxy {
     )
     let date, proxyAuthenticationToken, xenstoreData
     try {
+      if (networkId !== undefined) {
+        await Promise.all([
+          ...vm.VIFs.map(vif => xapi.deleteVif(vif)),
+          xapi.createVif(vm.$id, networkId),
```

Comment:
What if the VM is later deleted because the deployment failed?

Reply:
I don't see any deletion of VM VIFs on deleting the VM using the UI.

Reply:
Indeed, do you think it's an issue with our VM deletion?

Or what does this mean?

Reply:
I supposed that it's done automatically by `xenServer` i need to test it to be sure.

Reply:
No, you're right, XenAPI deletes the VIF if either the VM or the network are destroyed :+1: